pinot-segment-local/src/test/java/org/apache/pinot/segment/local/utils/SegmentOperationsThrottlerTest.java

```diff
@@ -308,11 +308,13 @@ public void testServingQueriesDisabledWithAcquireRelease()
               ? Integer.parseInt(
               CommonConstants.Helix.DEFAULT_MAX_SEGMENT_STARTREE_PREPROCESS_PARALLELISM_BEFORE_SERVING_QUERIES)
               : Integer.parseInt(CommonConstants.Helix.DEFAULT_MAX_SEGMENT_DOWNLOAD_PARALLELISM_BEFORE_SERVING_QUERIES);
+      // Default is too high: Integer.MAX_VALUE, take a large number of permits to ensure we don't block
```

Comment:
nit: `..., take a limited number of permits so that the test doesn't take too long to finish`

Reply:
done